core/src/main/java/com/novoda/noplayer/internal/exoplayer/PlayedAdverts.java

```diff
@@ -5,21 +5,25 @@
 
 import java.util.List;
 
-final class SkippedAdverts {
+final class PlayedAdverts {
 
-    private SkippedAdverts() {
+    private PlayedAdverts() {
         // Utility class.
     }
 
     static AdPlaybackState from(long currentPositionInMillis, List<AdvertBreak> advertBreaks, AdPlaybackState adPlaybackState) {
         AdPlaybackState adPlaybackStateWithSkippedAdGroups = adPlaybackState;
-        for (int i = advertBreaks.size() - 1; i >= 0; i--) {
-            AdvertBreak advertBreak = advertBreaks.get(i);
+        int numberOfAdvertBreaks = advertBreaks.size() - 1;
+        for (int advertBreakIndex = numberOfAdvertBreaks; advertBreakIndex >= 0; advertBreakIndex--) {
+            AdvertBreak advertBreak = advertBreaks.get(advertBreakIndex);
             if (advertBreak.startTimeInMillis() >= currentPositionInMillis) {
                 continue;
             }
 
-            adPlaybackStateWithSkippedAdGroups = adPlaybackStateWithSkippedAdGroups.withSkippedAdGroup(i);
```

Comment:
Why can't we skip the whole group but we need to mark each advert in the group as played instead?

Reply:
Yes we can actually. I'll revert the change.